app/services/puppet_fact_parser.rb

```diff
@@ -40,14 +40,20 @@ def operatingsystem
     end
     if os.description.blank?
       if os_name == 'SLES'
-        os.description = os_name + ' ' + orel.gsub!('.', ' SP')
+        os.description = os_name + ' ' + orel.gsub('.', ' SP')
```

Comment:
I'm not sure I understand why this change. It doesn't seem to change the behavior as we're not using `orel` anywhere after this anyway. gsub! (in place) should be slightly faster. If we're just going to make one substitution `orel['.'] = ' SP'` [should be even faster](https://github.com/JuanitoFatas/fast-ruby#stringsub-vs-stringgsub-vs-string-code) (petty, I know)